geoip: Install geolite database by default

bendlas opened the pull request:
This makes the geoiplookup command work without special parameters and
it establishes a directory that is mentioned in the man page.
(/nix/store/_geoip_/share/GeoIP/)

@7c6f434c

bendlas commented:
Apparently, this commit causes a couple of unnessecary rebuilds for programs, that depend on just the library. The way around this would be to split the geoiplookup commands out from the library and have only them depend on the database. Would that be desirable?

nckx commented:
> Apparently, this commit causes a couple of unnessecary rebuilds for programs, that depend on just the library.

Indeed. Packages that don't require a database really shouldn't (indirectly) depend on any. The GeoLite databases are quite large (50MiB) and change far more frequently than the library.

> The way around this would be to split the geoiplookup commands out from the library and have only them depend on the database. Would that be desirable?

I'm not sure I follow. I'd rather not split libraries and tools packaged upstream, if that's what you mean...

bendlas commented:
@nckx you're right, the more frequent rebuilds are a problem.
I've added a separate package geoip-full (with a configurable, but default database), so that nothing changes for dependents on the library.

nckx commented on pkgs/development/libraries/geoip/default.nix:
```diff
@@ -1,4 +1,6 @@
-{ stdenv, fetchurl }:
+# in geoip-database, you can insert a package defining ${geoip-database}/share/GeoIP
+# e.g. geolite-legacy
+{ stdenv, fetchurl, pkgs, geoip-database ? null }:
```
Nix convention is to use camelCase for this: `geoipDatabase`.

nckx commented on pkgs/top-level/all-packages.nix:
```diff
@@ -5767,6 +5767,10 @@ let
 
   geoclue2 = callPackage ../development/libraries/geoclue/2.0.nix {};
 
+  geoip-full = makeOverridable (callPackage ../development/libraries/geoip) {
```
Here too. Something like `geoipWithGeoliteDatabase` would be more conventional (and specific).

nckx commented:
Looks good to me!

A few minor comments in-line. Could you squash all commits into one and force-push?

bendlas committed:
geoip: add geoipWithDatabase for shell use

This installs (by default) geolite-legacy as the default database for geoiplookup

bendlas commented:
Done. I've additionally made the derivation name configurable, and `geoipWithDatabase` is called `geoip-tools`

bendlas commented:
Hey, can we get this change in?

nckx commented:
Certainly. Lost track of this; thanks for reminding me.

bendlas commented:
np, thanks!
